shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API.hs

```diff
@@ -1,12 +1,30 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableSuperClasses #-}
+
 -- | API to the Shelley ledger
 module Shelley.Spec.Ledger.API
   ( module X,
+    ShelleyBasedEra,
   )
 where
 
+import Cardano.Ledger.Era (Crypto)
+import Cardano.Ledger.Shelley (ShelleyBased, ShelleyEra)
 import Shelley.Spec.Ledger.API.ByronTranslation as X
 import Shelley.Spec.Ledger.API.Mempool as X
 import Shelley.Spec.Ledger.API.Protocol as X
 import Shelley.Spec.Ledger.API.Types as X
 import Shelley.Spec.Ledger.API.Validation as X
 import Shelley.Spec.Ledger.API.Wallet as X
+
+class
+  ( TPraosCrypto (Crypto era),
+    ShelleyBased era,
+    GetLedgerView era,
+    ApplyBlock era,
+    ApplyTx era
+  ) =>
+  ShelleyBasedEra era
```

Comment:
I'm open to suggestions for a better name and/or a better place for it.

Reply:
Is this top level of bundling especially useful? I feel like it'll be potentially confusing to have so many different variants on this...

Reply:
IMO, very. It is *the* constraint that consensus uses on `era`, as it implies everything we need. See for example: https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Block.hs.

Also, this gives us a single check to verify whether an era satisfies everything that is needed. (If you have to provide 3-4 instances, it easier to forget to provide one of them, e.g., `ApplyTx`.)

I could see that the name is confusing (compared to `ShelleyBased`), so I'm certainly open to a better name.

Reply:
OK. I'm also strugging to come up with a good name...

Reply:
`ShelleyMinimum`? I struggle as well...